plugins/server/io.github.flaxoos/ktor-server-kafka-jvm/2.0/documentation.md

```diff
@@ -0,0 +1,123 @@
+Integrate Apache Kafka into your Ktor applications with this plugin. 
+It provides flexible configuration options, admin client functionality, easy producer and consumer setup, and built-in Avro4k support. 
+Simplify event-driven communication and real-time data processing within your Ktor projects.
+
+## Usage
+
+The plugin provides a DSL that enables comprehensive Kafka configuration, adhering to the classes and properties defined in [org.apache.kafka.common.config](https://kafka.apache.org/21/javadoc/index.html?org/apache/kafka/common/config/package-summary.html), the DSL offers a fluent, programmatic way to set up your Kafka settings right within your Ktor application.
+
+- Setting up clients:
+```kotlin
+install(Kafka) {
+     schemaRegistryUrl = listOf("my.schemaRegistryUrl")
+     common { // <-- Define common configs
+         // properties
+     } 
+     admin { // <-- Creates an admin client
+         // properties
+     }
+     producer { // <-- Creates a producer
+         // properties
+     }
+     consumer { // <-- Creates a consumer
+         clientId = "my-client-id-override" //<-- Override common properties
+     }
+}
+```
+
+- Handling records
+```kotlin
+// in the plugin install block
+    val myTopic = TopicName.named("my-topic")
+    consumerConfig {
+        consumerRecordHandler(myTopic) { record ->
+            myService.save(record)
+        }
+    }
+```
+
+- Registering Schemas:
+
+```kotlin
+// in the plugin install block
+    registerSchemas {
+        using { // <-- optionally provide a client, by default CIO is used
+            HttpClient()
+        }
+        MyRecord::class at myTopic // <-- Will register schema upon startup
+    }
+```
+
+- Producing records
+```kotlin
+fun Application.sendKafkaEvent(event: MyEvent) {
+   kafkaProducer?.send(ProducerRecord("events", event.time.toString(), event.toRecord()))
+        ?.get(100, TimeUnit.MILLISECONDS)
+}
+```
+
+- Using the admin
+```kotlin
+val Application.kafkaMetrics
+    get() = kafkaAdminClient?.metrics()
+```
+
+### File Configuration:
+
+You can also easily install the Kafka plugin using an application configuration file, instead of configuring in code:
+
+```kotlin
+install(FileConfig.Kafka) {
+    // consumerConfig and registerSchemas are still done here 
+}
+```
+You can also specify a different path if needed:
+```kotlin
+install(FileConfig.Kafka("ktor.my.kafka")) {
+    // consumerConfig and registerSchemas are still done here
+}
+```
+Example file configuration:
+
+```hocon
```

Comment:
You could also include this config as a template, similar to here https://github.com/ktorio/ktor-plugin-registry/tree/main/plugins/server/io.ktor/auth-jwt/2.0

However, our templating engine doesn't currently omit the "code" configuration (i.e. install.kt) when HOCON / YAML is selected, so this might introduce a conflict here.  I'll look into it since it seems like a fairly important detail for honouring the configuration choice.

Reply:
I could also just put the
```
    consumerConfig {
        consumerRecordHandler("my-topic") { record ->
            myService.save(record)
        }
    }
    registerSchemas {
        MyRecord::class at named("my-topic") // <-- Will register schema upon startup
    }
```
part in the install and the rest in app.yaml, but maybe let's go ahead and merge this for now so it's available